ecms_base/modules/custom/ecms_workflow/src/EcmsWorkflowBundleCreate.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_workflow;
+
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Class EcmsWorkflowBundleCreate.
+ *
+ * Handles configuration updates when a new content type is added.
+ *
+ * @package Drupal\ecms_workflow
+ */
+class EcmsWorkflowBundleCreate {
+
+  use StringTranslationTrait;
+
+  /**
+   * The role id for content publishers.
+   */
+  const CONTENT_PUBLISHER_ROLE = 'content_publisher';
+
+  /**
+   * The role id for content authors.
+   */
+  const CONTENT_AUTHOR_ROLE = 'content_author';
+
+  /**
+   * The machine id for the default workflow.
+   */
+  const WORKFLOW_ID = 'editorial';
+
+  /**
+   * The entity_type.manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  private $entityTypeManager;
+
+  /**
+   * EcmsWorkflowBundleCreate constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity_type.manager service.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Grant node permissions to the author and publisher roles.
+   *
+   * @param string $contentType
+   *   The machine name of the new content type.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  private function setRolePermissions(string $contentType): void {
+    $storage = $this->entityTypeManager->getStorage('user_role');
+
+    $content_publisher_role = $storage->load(self::CONTENT_PUBLISHER_ROLE);
+
+    // Guard against an empty role.
+    if (empty($content_publisher_role)) {
+      return;
+    }
+
+    // Content Publisher role has all editing permissions.
+    $content_publisher_role->grantPermission("create {$contentType} content");
+    $content_publisher_role->grantPermission("edit any {$contentType} content");
+    $content_publisher_role->grantPermission("delete any {$contentType} content");
+
+    try {
+      $content_publisher_role->save();
+    }
+    catch (EntityStorageException $e) {
+      return;
+    }
+
+    $content_author_role = $storage->load(self::CONTENT_AUTHOR_ROLE);
+
+    // Guard against an empty role.
+    if (empty($content_author_role)) {
+      return;
+    }
+
+    // Content Author role has limited editing permissions.
+    $content_author_role->grantPermission("create {$contentType} content");
+    $content_author_role->grantPermission("edit own {$contentType} content");
+
+    try {
+      $content_author_role->save();
+    }
+    catch (EntityStorageException $e) {
+      return;
+    }
+  }
+
+  /**
+   * Create a new user for use with the API.
```

Comment:
This method description doesn't seem right.

Reply:
Ah yes! That would be someone else's comment :)